### PR TITLE
Update GH tests action to use pytest instead of nosetests

### PR DESCRIPTION
**Pull request by dbarnett**
Adds py 3.12 to the matrix and drops the oldest one for Windows py 3.4.

### CODE REVIEW

**Comment by dbarnett** [comment]
As discussed in https://github.com/google/vimdoc/pull/134#discussion_r1735043091, still planning to update the supported python versions once the groundwork is laid.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -27,9 +28,8 @@ jobs:
             os: ubuntu-20.04
           - py: "3.5"
             os: macos-12
-          - py: "3.4"
-            os: windows-2019
-            installer: easy_install
+    env:
+      PIP_TRUSTED_HOST: ${{ matrix.py == '3.5' && 'pypi.python.org pypi.org files.pythonhosted.org' || '' }}
```
Comment (malcolmr) [inline .github/workflows/tests.yml:32]: Could you add a comment to explain why we need to do this for Python 3.5? (I see we did it unconditionally before; I assume it's not needed for more-recent versions of Python?).

**Review by malcolmr — APPROVED** [review]
LGTM

File: .github/workflows/tests.yml (hunk 1 above)
Comment (dbarnett) [reply .github/workflows/tests.yml:32]: Sure, but fwiw I'm going to be following up with another change right after this to bump min supported py version and remove the workaround again. This was just the easiest way to break up small self-contained changes for testing vs bigger stuff. 

The short version is just that pip is newer than legacy setuptools/easy_install and really really old systems seem to not have it working correctly.

Comment (dbarnett) [inline .github/workflows/tests.yml:32]:
```suggestion
      # Workaround for pypi rejecting older SSL versions from older py versions: 
      # https://stackoverflow.com/questions/25981703/pip-install-fails-with-connection-error-ssl-certificate-verify-failed-certi
      PIP_TRUSTED_HOST: ${{ matrix.py == '3.5' && 'pypi.python.org pypi.org files.pythonhosted.org' || '' }}
```

Comment (dbarnett) [reply .github/workflows/tests.yml:34]: Oh sorry, misread this and thought you were asking why easy_install was conditionally needed. Added a comment for trusted hosts. And I saw there may be other better workarounds when I checked that link again, but also for just a test runner action I feel like the stakes are fairly low for that kind of security risk.

**Comment by malcolmr** [comment]
LGTM, thanks. You can merge this yourself, I believe?

**Comment by dbarnett** [comment]
> LGTM, thanks. You can merge this yourself, I believe?

Didn't think I could. I can't right now, but it's possible that's just because the notify-discord action failed.

**Comment by malcolmr** [comment]
> Didn't think I could. I can't right now, but it's possible that's just because the notify-discord action failed.

I guess I still don't understand GitHub's ACL model. Merged, thanks!

**Pull request merged**